cpp/src/parquet/encoding.cc

```diff
@@ -1156,16 +1160,21 @@ inline int PlainBooleanDecoder::DecodeArrow(
 
 int PlainBooleanDecoder::Decode(uint8_t* buffer, int max_values) {
   max_values = std::min(max_values, num_values_);
-  bool val;
+  constexpr int kBatchSize = 1024;
+  std::array<bool, kBatchSize> bit_read_scratch;
   ::arrow::internal::BitmapWriter bit_writer(buffer, 0, max_values);
```

Comment:
In fact I don't know the differences between bitWriter and `FirstTimeBitWriter`... Would the later one be better? 🤔

Reply:
And in fact the fastest the way is directly copy LSB bits to BitWriter 🤔